File: frontend/src/types/InspectionResult.ts
export type ResultStatusValue = "NORMAL" | "ABNORMAL";

export interface InspectionResult {
  id: number;
  task_id: number;
  device_id: number;
  device_code?: string | null;
  item_code: string;
  item_name?: string | null;
  result_status: ResultStatusValue | string;
  measured_value: string;
  photo_url: string;
  note: string;
  ticket_id?: number | null;
  updated_at?: string | null;
  task_status?: string;
  building_id?: number;
}

File: backend/src/models/inspection_result.py
from datetime import datetime

from sqlalchemy import UniqueConstraint
from sqlalchemy.orm import Mapped, mapped_column

from src.config.database import Base


class InspectionResult(Base):
    __tablename__ = "inspection_result"
    __table_args__ = (
        UniqueConstraint("task_id", "device_id", "item_code", name="uq_result_task_device_item"),
    )

    id: Mapped[int] = mapped_column(primary_key=True)
    task_id: Mapped[int]
    device_id: Mapped[int]
    item_code: Mapped[str]
    result_status: Mapped[str]
    measured_value: Mapped[str] = mapped_column(default="")
    photo_url: Mapped[str] = mapped_column(default="")
    note: Mapped[str] = mapped_column(default="")
    updated_at: Mapped[datetime | None] = mapped_column(nullable=True)
